### PR TITLE
feat(oui-progress): progress component implementation

### PR DESCRIPTION
**Pull request by ganeshkumar1989**
UU-842
Added progress component

Staging link: http://feature_oui_progress_build.ui-kit.ovh/#!/oui-angular/progress

**Linked To:**
https://github.com/ovh-ux/ovh-ui-kit/pull/213
https://github.com/ovh-ux/ovh-ui-kit-documentation/pull/112

### CODE REVIEW

**Comment by AxelPeter** [comment]
@ganeshkumar1989 Is there a staging ?

**Comment by ganeshkumar1989** [comment]
I have added the staging link @AxelPeter . The link is alos available in the ticket

File: packages/oui-progress/README.md [hunk 1]
```diff
@@ -0,0 +1,111 @@
...
+<oui-progress type="warning" indeterminate="true"></oui-progress>
+```
+
+## API
+
+### oui-progress
+
+| Attribute               | Type            | Binding | One-time binding | Values                                 | default | Description                            |
+| ----                    | ----            | ----    | ----             | ----                                   | ----    | ----                                   |
+| type                    | String          | @       | yes              | `info`, `success`, `warning`, `error`  |         | Type of the progress component       |
```
Comment (AxelPeter) [inline packages/oui-progress/README.md]: From the specs, this attribute should be only on children

File: packages/oui-progress/README.md [hunk 2]
```diff
@@ -0,0 +1,111 @@
...
+| max-value               | Number          | @?      | yes              |                                        | 100     | The max value for the progress bar    |
+| indeterminate           | Boolean         | <?      | yes              |                                        | false   | Enable/Disable indeterminate mode. When in this mode, the progress component shows an indefinite wait and the max-value would not have any effect. In this mode, the progress component should not have any other component in it |
```
Comment (AxelPeter) [inline packages/oui-progress/README.md]: Guidelines has changed since, this is not needed

File: packages/oui-progress/README.md [hunk 3]
```diff
@@ -0,0 +1,111 @@
...
+
+### oui-progress-bar
+
+| Attribute               | Type            | Binding | One-time binding | Values                                 | default       | Description                            |
+| ----                    | ----            | ----    | ----             | ----                                   | ----          | ----                                   |
+| type                    | String          | @?      | yes              | `info`, `success`, `warning`, `error`  | parent's type | Type of the progress bar. If not provided, takes the type of the parent |
+| value                   | Number          | <       | no               |                                        |               | The current value for the progress bar                                  |
+| label-align             | String          | @?      | yes              | `left`, `right`                        | right         | The alignment of the label inside the progress bar                            |
```
Comment (AxelPeter) [inline packages/oui-progress/README.md]: No need, the value is always right aligned (see UI specs)

File: packages/oui-progress/src/progress-bar.controller.js [hunk 4]
```diff
@@ -0,0 +1,23 @@
...
+        addDefaultParameter(this, "type", this.parent.type);
+        addDefaultParameter(this, "labelAlign", "right");
+    }
+
+    $postLink () {
+        const maxValuePercentage = 100;
+        const widthPrecentage = (this.value * maxValuePercentage) / this.parent.maxValue;
+        this.$element.addClass("oui-progress__bar");
+        this.$element.addClass(`oui-progress__bar_${this.type}`);
+        this.$element.css("width", `${widthPrecentage}%`);
```
Comment (AxelPeter) [inline packages/oui-progress/src/progress-bar.controller.js]: DOM manipulations need a $timeout wrap

File: packages/oui-progress/src/progress-threshold.controller.js [hunk 5]
```diff
@@ -0,0 +1,19 @@
...
+
+    $onInit () {
+        throwErrorOnMissingAttributeValue(this.$element[0], "value");
+    }
+
+    $postLink () {
+        const maxValuePercentage = 100;
+        const leftPrecentage = (this.value * maxValuePercentage) / this.parent.maxValue;
+        this.$element.addClass("oui-progress__threshold");
+        this.$element.css("left", `${leftPrecentage}%`);
```
Comment (AxelPeter) [inline packages/oui-progress/src/progress-threshold.controller.js]: Same as above, $timeout :)

File: packages/oui-progress/README.md (hunk 1 above)
Comment (ganeshkumar1989) [reply packages/oui-progress/README.md]: The parent and child are now combined to a single component

File: packages/oui-progress/README.md (hunk 2 above)
Comment (ganeshkumar1989) [reply packages/oui-progress/README.md]: This has been removed

File: packages/oui-progress/README.md (hunk 3 above)
Comment (ganeshkumar1989) [reply packages/oui-progress/README.md]: This is removed :)

File: packages/oui-progress/src/progress-bar.controller.js (hunk 4 above)
Comment (ganeshkumar1989) [reply packages/oui-progress/src/progress-bar.controller.js]: This is done :)

File: packages/oui-progress/src/progress-threshold.controller.js (hunk 5 above)
Comment (ganeshkumar1989) [reply packages/oui-progress/src/progress-threshold.controller.js]: This is done :)

**Comment by jleveugle** [comment]
Hello @ganeshkumar1989,
Your pull request is now closed due to inactivity.
Feel free to re-open it if you have any update.
Anna Bot for OVH UX <3

**Comment by ovh-cds** [comment]
CDS Report yarn-build#922.0 ✘  
* build &amp; test
  * build ✔ 
  * test ✘

**Comment by ovh-cds** [comment]
CDS Report yarn-build#923.0 ✘  
* build &amp; test
  * build ✘  
  * test ✔

**Pull request merged**